### DISCUSSION TITLE
Users - introduce yourself and your applications!

### DISCUSSION THREAD

**Discussion by isukendall** [kthorp/pyfao56#56] [General]
There are many new users of pyfao56 with various applications. Knowing what others are working on will encourage collaboration and help track users and applications.

If you like, **please introduce yourself in this thread**, and feel free to include any relevant details such as: 

- Your lab or institution
- Your team members
- Your application and products
- Relevant websites or publications

**Comment by isukendall** [comment]
I'll go first.

My name is Kendall DeJonge, I am a Research Agricultural Engineer with the USDA-ARS Water Management and Systems Research Unit in Fort Collins, Colorado. Our research program focuses on Limited Irrigation management and evapotranspiration measurement and modeling. For over a decade we had used a spreadsheet model for ET and water balance, but the number of individual treatments and replicates made it overwhelming and difficult to repeat. We had discussed for a long time the need for a programmable version of our model. 

My team began working with Kelly Thorp to develop pyfao56 v1.1.0, to expand on the initial version of the tool and integrate some of the nuanced approaches of our own water balance model. Josh Brekel @jbrekel was an initial developer supporting v1.1.0 and v1.2.0, and Tyler Pokoski @tpokoski has continued work toward v1.3.0 and beyond. Our efforts have been focused on development of pyfao56 itself, as well as additional development and customization for our unique needs at the Limited Irrigation Research Farm (LIRF). 

Our customization of pyfao56 for the Limited Irrigation Research Farm (LIRF) resulted in the attached paper, which serves as a great example of how the tool can be customized for various uses. We continue to work with Kelly and others on new functionality, and we're excited that so many people are interested in using pyfao56. 


[DeJonge - 2024 - AWM - pyfao56 LIRFIP.pdf](https://github.com/user-attachments/files/16503182/DeJonge.-.2024.-.AWM.-.pyfao56.LIRFIP.pdf)

**Reply by jbrekel** [reply]
Hi everyone, 

I wanted to add on to Kendall's post to say that although I no longer work on ET research in any official capacity, I am still keenly watching the development of pyfao56 from afar. I think I understand the code base pretty well, so feel free to reach out with any questions that you think I could help with.  I care about this project and I want to see it continue to succeed in the future. I am happy to see these discussions forums launch!

If you want to get into contact with me, visit my Github profile or [my website](joshbrekel.com).

---Josh

**Reply by tpokoski** [reply]
Hello!

As Kendall mentioned, I am the Engineering Technician on his team. I mostly work at the Limited Irrigation Research Farm (LIRF) in the field, and work on our LIRFIP repository when I can. 

My job title might fool you, I am in fact not an engineer. My background consists of some bachelor's degrees in Biology, Environmental Science, and Botany, as well as a master's in GIS and Spatial Data Management. Coding experience started of with R in my undergrad, and then transitioning to Python with my grad school. 

Most of our coding focus can be found in the paper that Kendall linked, but I am also interested in helping add features to the base pyfao56.

**Comment by ansleybrown1337** [comment]
Hi Everyone,

My name is A. J. Brown, and I am an Agricultural Data Scientist with the Colorado State University Agricultural Water Quality Program.  Our program works directly with producers and researchers around the state to research and develop best management practices that protect surface and ground water quality.  A large part of this effort is driven my irrigation water management and subsequently quantifying ET.

I do a fair bit of coding, and have followed pyfao56 since it's initial release, but am now starting to use it in a collaborative effort with USDA-ARS LIRF and other CSU researchers to improve an already existing irrigation scheduling software called [WISE](https://erams.com/catena/tools/agricultural-resources/irrigation-advisor/).  Part of this improvement will be integrating pyfao56 into the model for improved ET estimation.  The project is funded by USDA-NIFA.

Additionally, I would be interested in contributing to code improvements as possible, and creating forks for other use cases as needed.  For example, I need to calculate 5-minute ET values based on weather station data instead of the current daily or hourly values inherent in existing pyfao56 functions.

Unfortunately, I may not be able to attend the inaugural meeting on this coding effort, but I hope to join in the future.

**Comment by pahlse** [comment]
Hi all,

My name is Elias Pahls, and I'm an ag student at Wageningen University in the Netherlands. I'm currently doing my MSc. thesis on Rice-Wheat cropping scenarios in North-West India following the FAO56 methodology. I've been using CROPWAT 8.0 and stumbled across this repo a few weeks ago. Thanks for the great work! 

Now looking into adding a paddy rice functionality to the existing code to simulate ponded conditions. Lets see what comes of it. Bummer that I missed the last quarterly meeting, but will try to attend in Feb.

Cheers till then!

**Reply by kthorp** [reply]
Hi Elias and thanks for the introduction.  I was actually in northern India earlier this year for the Roorkee Water Conclave at IIT Roorkee.  I presented a few slides on pyfao56 there.  Saw some rice, wheat, and sugarcane systems and flood irrigation methods there. Best wishes for a successful M.S. thesis.  If you develop pyfao56 methods that would be useful to the community, I'd welcome pull requests to bring those methods into the package.

**Reply by pahlse** [reply]
Hi Kelly,
Great to hear that you were in Roorkee, I hope you enjoyed! I was there myself a few weeks ago.
I forked the repo and let rip (see here: [pahse/pyfao56-rice](https://github.com/pahlse/pyfao56-rice/tree/main)). Still quite a few things to iron out, but core concepts are there. I'm trying to stick to CROPWAT 8.0 in terms of user inputs and following the original excel sheet (courtesy my professor) from 2003 that formed the basis of the rice part of CROPWAT. 
Take a look if you find the time.

**Reply by kthorp** [reply]
Looks like you have a good start.  Thank you for your efforts. I'm interested to know how it turns out.

**Comment by isukendall** [comment]
Hi Elias, thanks for reaching out! Glad you have found pyfao56 to be useful to your research. It certainly would be interesting to learn more about your project, we hope you can make it to the next meeting! Ponded conditions is an interesting scenario, curious how you plan to tackle it.

Please keep us updated with your progress and reach out if you have any questions.


Kendall DeJonge
Agricultural Engineer
USDA-ARS

**Reply by pahlse** [reply]
Hi Kendall,
Thanks for the reply.
I'm trying to assess the SWB components of three different cropping schemes common to the region. Direct-Seeded rice is heavily promoted here as a water-saving technology as compared to "water-guzzling" paddy rice. But once you get into the details nobody really knows how and if it actually saves water, especially with varying soil types, and if ETa is not reduced or even increased depending on planting dates and crop varieties. My hypothesis is that observed irrigation savings are due to reduced percolation, and with 95% of irrigation water coming from groundwater here I'd consider that a return flow and not a real water saving in a strict sense; i.e. a "wet water saving". I'm hoping to lay out the details of three idealized systems using the pyfao56 package to showcase the complexity and dynamics at play when switching from paddy rice to direct seeded rice.

I'll let you know if anything comes up.
Cheers!

**Comment by pahlse** [comment]
Hi All,

I'm posting here again to invite you to my MS thesis presentation, where I'll be presenting my adaptations to the pyfoa56 model for paddy rice systems. The goal of the thesis was to quantify how different rice establishment methods (TPR, DSR, AWD) affect real water savings (ET) vs. apparent savings (irrigation reductions) in north-west India’s groundwater dependent rice-wheat systems. The approach worked great for modeling the plot-level soil water balance and showing distinct differences between the methods modeled.

Key adaptations to the pyfao56 code:
- Adapted the `model` class to track standing water table, soil moisture above field capacity and residual moisture.
- Incorporated van Genuchten hydraulic conductivity based on volumetric water content to limit percolation for irrigation above field capacity
- Added a `landprep` class to model plow sole establishment and Ksat reduction before transplanting

The presentation will be held on April 17th. Please let me know if your interested in joining, and I will send you a seperate email with the official invitation.

All the best,
Elias

**Reply by isukendall** [reply]
Hello Elias, thanks for the update. This sounds like very interesting work. I would love to attend your MS thesis presentation to learn more, please send the invite to kendall.dejonge@usda.gov. Good luck!

**Reply by jbrekel** [reply]
Hi Elias, 

This sounds interesting, and I would like to attend if I can. Please send the link to josh.brekel@gmail.com. You'll do great! You are always the expert at your own defense. :)

Best, 

Josh

**Comment by sumondatta** [comment]
Elis, please share the invite with me as well. Thanks.

From: Kendall DeJonge ***@***.***>
Reply-To: kthorp/pyfao56 ***@***.***>
Date: Tuesday, March 25, 2025 at 9:27 AM
To: kthorp/pyfao56 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [kthorp/pyfao56] Users - introduce yourself and your applications! (Discussion #56)


Hello Elias, thanks for the update. This sounds like very interesting work. I would love to attend your MS thesis presentation to learn more, please send the invite to ***@***.******@***.***>. Good luck!

**Comment by kthorp** [comment]
Hi Elias,  Please include me in the invite as well. It seems you have made rapid progress in a short time!  ~Kelly
________________________________

**Comment by Amhashem1** [comment]
Thank you Elias, and best of luck with your dissertation. Please add me to
the invitation list. Here is my email address ***@***.***

Ahmed
